wingtips-spring-webflux/src/main/java/com/nike/wingtips/spring/webflux/client/WingtipsSpringWebfluxExchangeFilterFunction.java

```diff
@@ -66,15 +56,33 @@
  *         .attribute(TracingState.class.getName(), tracingState)
  *         .exchange()
  *         </pre>
- *         NOTE: If the current thread's tracing state is setup when you execute the WebClient call *and*
- *         you've explicitly specified {@link TracingState} in the request attributes, and those two tracing states
- *         are different, then the request attribute's tracing state will take precedence for the purpose of the
- *         WebClient call.
+ *     </li>
+ *     <li>
+ *         UNRELIABLE - Ensure your current thread's tracing state is setup when you execute the WebClient call
+ *         (i.e. when you call the WebClient's {@link
+ *         org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec#exchange() exchange()} or
+ *         {@link org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec#retrieve() retrieve()}
+ *         method). This is done via the usual Wingtips {@code Tracer.getInstance().start*(...)} methods. If you're
+ *         using the {@link com.nike.wingtips.spring.webflux.server.WingtipsSpringWebfluxWebFilter
+ *         WingtipsSpringWebfluxWebFilter} for handling things on the server side, and you've made sure to propagate
+ *         your tracing state during any thread hopping, then this should already be setup for you.
+ *
+ *         <p><b>WARNING: This mechanism is unreliable</b>, because different versions of the Spring WebFlux WebClient
+ *         have different threading behaviors when filters are executed. Sometimes filters are executed on the same
+ *         thread (in which case this mechanism would work), but other times filters are executed on different threads
+ *         (in which case this mechanism would fail). Since you cannot predict the threading behavior that gets used
+ *         as you change Spring versions (they do not consider threading behavior changes like this to be breaking
+ *         changes), you should use the request attribute mechanism described above instead.
+ *
+ *         <p>NOTE: If the current thread's tracing state is setup when you execute the WebClient call (and you're
+ *         lucky enough to have the filter execute on the same thread) *and* you've explicitly specified
+ *         {@link TracingState} in the request attributes, and those two tracing states are different, then the
+ *         request attribute's tracing state will take precedence for the purpose of the WebClient call.
```

Comment:
Spring changed their threading behavior for the `WebClient`'s exchange filter functions in a more recent version, breaking this automagic tracing state propagation. So now the only officially supported way to propagate wingtips tracing state through `WebClient` is to use the preferred method described above (the `.attribute(TracingState.class.getName(), tracingState)` mechanism).